WIP: Improvement - New block - How To Schema

AbhijitNage123 opened the pull request:
### Description
<!-- Please describe what you have changed or added -->
Improvement - New block - How To Schema

### Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change -->
New feature (non-breaking change which adds functionality)

### Checklist:
- [x] My code is tested
- [x] My code follows accessibility standards <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included any necessary tests <!-- if applicable -->
- [x] I've included developer documentation <!-- if applicable -->
- [x] I've added proper labels to this pull request <!-- if applicable -->

vrundakansara commented on classes/class-uagb-config.php:
```diff
@@ -84,6 +84,121 @@ public static function get_block_attributes() {
+						'description' => __( 'How To Configure How-To Schema In Ultimate Addons For Gutenberg?', 'ultimate-addons-for-gutenberg' ),
+						'default'     => true,
+						'attributes'  => array(
+							'classMigrate'            => false,
```
Do we need this?

AbhijitNage123 replied:
Removed. Done!

vrundakansara commented on config/webpack.config.dev.js:
```diff
@@ -56,8 +56,8 @@ fs.readdir( paths.pluginSrc + "/blocks", function(err, items) {
 			let file_path = result.stats.entry
-			let new_path = file_path.replace( paths.pluginSrc + "/blocks/", "" );
-			new_path = new_path.replace( "/style.scss", "" );
+			let new_path = file_path.replace( paths.pluginSrc + "\\blocks\\", "" );
```
Do we need this?

AbhijitNage123 replied:
Removed.

vrundakansara commented on config/webpack.config.prod.js:
```diff
@@ -61,9 +61,9 @@ fs.readdir(paths.pluginSrc + "/blocks", function(err, items) {
 			let file_path = result.stats.entry
 
-			let new_path = file_path.replace( paths.pluginSrc + "/blocks/", "" );
+			let new_path = file_path.replace( paths.pluginSrc + "\\blocks\\", "" );
```
Do we need this?

AbhijitNage123 replied:
Removed

vrundakansara commented on src/blocks/How-To/editor.scss:
```diff
@@ -0,0 +1,12 @@
+.uagb-social-share__layout-horizontal .uagb-social-share__wrap .block-list-appender {
```
Do we need this?

AbhijitNage123 replied:
Updated CSS. Done!

vrundakansara commented on src/blocks/How-To/save.js:
```diff
@@ -0,0 +1,202 @@
+
+	if ( mainimage && mainimage.url ) {
+
+		image_icon_html = <img className="uagb-howto__source-image" src={url} />
```
Add title attribute to image tag.

AbhijitNage123 replied:
Added title to image tag. Done

RajkiranBagal commented on classes/class-uagb-block-helper.php:
```diff
@@ -16,6 +16,310 @@
+			// $t_selectors[" .uagb-tools .uagb-tools__label"] = array(
+			//    	'font-size' => UAGB_Helper::get_css_value( $attr['toolsFontSizeTablet'], $attr['materialsFontSizeType'] ),
+			//     'line-height' => UAGB_Helper::get_css_value( $attr['toolsLineHeightTablet'], $attr['materialsLineHeightType'] ),
+			// );
```
Remove this commented code.

AbhijitNage123 replied:
Removed

RajkiranBagal commented on classes/class-uagb-block-helper.php:
```diff
@@ -16,6 +16,310 @@
+			// $selectors[" span.uagb-howto__source-wrap .uagb-howto__source-image"] = array(
+			// 	"width"=> UAGB_Helper::generate_css( $attr['imgWidth'], "%" )
+			// );
+
```
Remove this commented code.

AbhijitNage123 replied:
Removed

RajkiranBagal commented on classes/class-uagb-config.php:
```diff
@@ -84,13 +84,127 @@ public static function get_block_attributes() {
 						'description' => __( 'This block gives you advanced options to insert a number of columns within a single row.', 'ultimate-addons-for-gutenberg' ),
 						'default'     => true,
 						'attributes'  => array(
-							'classMigrate'           => false,
```
Why removed this code from uagb/columns?

AbhijitNage123 replied:
Mistakenly, removed by me. Added again.

RajkiranBagal commented on src/blocks/How-To/attributes.js:
```diff
@@ -0,0 +1,441 @@
+/**
+ * BLOCK: UAGB How-To Attributes
+ */
+const ITEM_TOOLS_COUNT = 1
```
As this value is static, No need to declare a variable for it.

AbhijitNage123 replied:
Removed

RajkiranBagal commented on src/blocks/How-To/attributes.js:
```diff
@@ -0,0 +1,441 @@
+ */
+const ITEM_TOOLS_COUNT = 1
+
+const ITEM_MATERIALS_COUNT = 1
```
As this value is static, No need to declare a variable for it.

AbhijitNage123 replied:
Removed

RajkiranBagal commented on src/blocks/How-To/block.js:
```diff
@@ -0,0 +1,42 @@
+import UAGB_Block_Icons from "../../../dist/blocks/uagb-controls/block-icons"
+import attributes from "./attributes"
+import edit from "./edit"
+// import deprecated from "./deprecated"
```
Remove this.

AbhijitNage123 replied:
Removed

RajkiranBagal commented on src/blocks/How-To/block.js:
```diff
@@ -0,0 +1,42 @@
+	attributes,
+	edit,
+	save,
+	// deprecated,
```
Remove this, as well as delete the deprecated file

AbhijitNage123 replied:
Removed & deleted

RajkiranBagal commented on src/blocks/How-To/edit.js:
```diff
@@ -0,0 +1,1051 @@
+	constructor() {
+		super( ...arguments )
+
+		// this.splitBlock = this.splitBlock.bind( this )
```
Remove commented code.

RajkiranBagal commented on src/blocks/How-To/edit.js:
```diff
@@ -0,0 +1,1051 @@
+		let loadtoolsLoadGoogleFonts;
+		let loadmaterialsLoadGoogleFonts; 
+
+		if( headLoadGoogleFonts == true ) {
```
Use Yoda condition and check strictly with ===

RajkiranBagal commented on src/blocks/How-To/edit.js:
```diff
@@ -0,0 +1,1051 @@
+			)
+		}
+
+		if( subHeadLoadGoogleFonts == true ) {
```
Use Yoda condition and check strictly with ===

RajkiranBagal commented on src/blocks/How-To/edit.js:
```diff
@@ -0,0 +1,1051 @@
+			)
+		}
+
+		if( toolsLoadGoogleFonts == true ) {
```
Use Yoda condition and check strictly with ===

AbhijitNage123 replied:
Done

RajkiranBagal commented on src/blocks/How-To/edit.js:
```diff
@@ -0,0 +1,1051 @@
+			)
+		}
+
+		if( materialsLoadGoogleFonts == true ) {
```
Use Yoda condition and check strictly with ===

AbhijitNage123 replied:
Done

RajkiranBagal commented on src/blocks/How-To/edit.js:
```diff
@@ -0,0 +1,1051 @@
+			onReplace,
+			anchor,
+			//JSON data
+			schemaJsonData,
```
Remove this, if not using

AbhijitNage123 replied:
Needed

RajkiranBagal replied:
Where is this attribute used - schemaJsonData

AbhijitNage123 replied:
Right it's a prop. Removed

RajkiranBagal commented on src/blocks/How-To/save.js:
```diff
@@ -0,0 +1,207 @@
+			}
+	}
+
+	let image_icon_html = ""
```
Use single quotes for such declarations.

AbhijitNage123 replied:
Done

RajkiranBagal commented on src/blocks/How-To/styling.js:
```diff
@@ -0,0 +1,270 @@
+	var selectors = {
+		
+		" .block-editor-rich-text__editable.uagb-howto-desc-text": {
+			// "text-align": descriptionAlign,
```
Remove all commented and unwanted code.

AbhijitNage123 replied:
Done

RajkiranBagal commented:
Change block folder name from How-To to how-to

RajkiranBagal commented:
Run PHPCS and also pull next-release branch

RajkiranBagal commented on src/blocks/How-To/edit.js:
```diff
@@ -0,0 +1,1051 @@
+import memoize from "memize"
+import UAGB_Block_Icons from "../../../dist/blocks/uagb-controls/block-icons"
+import styling from "./styling"
+// import image from "../../../dist/blocks/image/placeholder"
```
Remove unwanted code and also unused imports

AbhijitNage123 replied:
Done

RajkiranBagal commented on src/blocks/How-To/edit.js:
```diff
@@ -0,0 +1,1051 @@
+	PanelBody,
+	SelectControl,
+	RangeControl,
+	ButtonGroup,
```
remove all the unused constants declared.

AbhijitNage123 replied:
Done

RajkiranBagal replied:
I can still see many used attr declared in this file as well as other files.

RajkiranBagal commented on src/blocks/How-To/edit.js:
```diff
@@ -0,0 +1,1051 @@
+				subHeadLineHeightTablet,
+				separatorWidth,
+				separatorWidthType,
+				seperatorStyle,
```
Remove all unwanted attributes

AbhijitNage123 replied:
Done

RajkiranBagal replied:
I can still see many used attr declared in this file as well as other files.

RajkiranBagal commented:
@AbhijitNage123 Can we add a separate function for each PanelBody Settings so that it will be easier to edit in the future. Refer FAQ

RajkiranBagal commented:
@AbhijitNage123 Discuss this Name and Description with Vrunda Ma'am and change it accordingly - https://imgur.com/wbisb8f

RajkiranBagal commented:
@AbhijitNage123 
Change this setting format to.
Select Image
Size
Width

RajkiranBagal commented:
@AbhijitNage123 Are these colors supposed to be apllied to these also - https://imgur.com/CuykaTQ

RajkiranBagal commented:
Total Time gets the side of image if image size is reduced - https://imgur.com/JTin5Xt

RajkiranBagal commented:
This should be conditional - https://imgur.com/yGFcRaU..When Time is OFF then its Margin settings should not be displayed

RajkiranBagal commented:
These settings are not applying anywhere - https://imgur.com/EjK4RDP

RajkiranBagal commented:
Tools Typography doesn't seem to work

RajkiranBagal commented:
Color setting does'nt seem to work - https://imgur.com/GPEibpT

RajkiranBagal commented:
Change these to "Tools" and "Materials" respectively - https://imgur.com/VnWwqoy

RajkiranBagal commented:
Cannot find a setting for styling of this line - https://imgur.com/CGmMMW2

RajkiranBagal commented:
@AbhijitNage123 Display proper notices for the required Schema inputs

AbhijitNage123 commented:
> Total Time gets the side of image if image size is reduced - https://imgur.com/JTin5Xt

Fixed

AbhijitNage123 commented:
> @AbhijitNage123
> Change this setting format to.
> Select Image
> Size
> Width

Done. also hide size & width option when the image is empty

AbhijitNage123 commented:
> This should be conditional - https://imgur.com/yGFcRaU..When Time is OFF then its Margin settings should not be displayed.
> 
> Same for cost

Done. Added condition

AbhijitNage123 commented:
> @AbhijitNage123 Are these colors supposed to be apllied to these also - https://imgur.com/CuykaTQ

Fixed

AbhijitNage123 commented:
> Cannot find a setting for styling of this line - https://imgur.com/CGmMMW2

Added common color option for secondary heading

AbhijitNage123 commented:
> Change these to "Tools" and "Materials" respectively - https://imgur.com/VnWwqoy

Change to Tools and Materials

AbhijitNage123 commented:
> These settings are not applying anywhere - https://imgur.com/EjK4RDP
> Are these settings supposed to be here?

Added common option for color

AbhijitNage123 commented:
> Tools Typography doesn't seem to work

Updated the code. fixed

AbhijitNage123 commented:
> @AbhijitNage123 Are these colors supposed to be apllied to these also - https://imgur.com/CuykaTQ

Added separate option for Secondory heading color

vrundakansara commented:
Do we need all these CSS files?
```
assets/css/blocks/how-to-tools-child.css
assets/css/blocks/how-to-steps-child.css
assets/css/blocks/how-to-materials-child.css
assets/css/blocks/how-to-child.css
assets/css/blocks/How-To.css
```

vrundakansara commented on classes/class-uagb-config.php:
```diff
@@ -84,6 +84,121 @@ public static function get_block_attributes() {
+						'description' => __( 'How To Configure How-To Schema In Ultimate Addons For Gutenberg?', 'ultimate-addons-for-gutenberg' ),
+						'default'     => true,
+						'attributes'  => array(
+							'classMigrate'            => false,
```
Do we need this?
`'classMigrate'            => false,`

AbhijitNage123 replied:
Not needed. Removed

vrundakansara commented:
`dist/blocks/image/placeholder.png`
Shift this to assets folder.

vrundakansara commented on classes/class-uagb-config.php:
```diff
@@ -84,6 +84,121 @@ public static function get_block_attributes() {
 					),
+					'uagb/how-to'   => array(
+						'slug'        => '',
+						'title'       => __( 'How-To' ),
```
This does not have the 2nd text domain parameter.

AbhijitNage123 replied:
Fixed Added text domain

vrundakansara commented on classes/class-uagb-block-helper.php:
```diff
@@ -16,6 +16,258 @@
+				'color' => $subHeadingColor,
+				),
+
+			' .uagb-howto__wrap span.uagb-howto__time-wrap .uagb-howto-timeNeeded-text,
```
Check this code by using multiple how-to blocks on the same page with different colors set for them.

AbhijitNage123 replied:
> @AbhijitNage123 Width Not working - https://imgur.com/IAtPyp5

Fixed

vrundakansara commented on classes/class-uagb-block-helper.php:
```diff
@@ -16,6 +16,258 @@
+					);
+
+			} else {
+				for($i=0;$i<$attr['tools_count'];$i++){
```
Use a common class, do not repeat the CSS.

AbhijitNage123 replied:
Removed loop and added common class

vrundakansara commented on classes/class-uagb-block-helper.php:
```diff
@@ -16,6 +16,258 @@
+					);
+
+			} else {
+				for($i=0;$i<$attr['material_count'];$i++){
```
Use common class.

AbhijitNage123 replied:
Done

vrundakansara commented on src/blocks/How-To/attributes.js:
```diff
@@ -0,0 +1,438 @@
+
+const materials = []
+
+for (var i = 1; i <= 1; i++) {
```
If these are just single values, why use for loop?

vrundakansara commented on src/blocks/How-To/attributes.js:
```diff
@@ -0,0 +1,438 @@
+		type: "boolean",
+		default: true
+	},
+	classMigrate: {
```
Do we need this variable?

AbhijitNage123 replied:
Removed

vrundakansara commented on src/blocks/How-To/attributes.js:
```diff
@@ -0,0 +1,438 @@
+		type: "string",
+		default: " USD"
+	},
+	headingId: {
```
Do we need this?

AbhijitNage123 replied:
Not needed. removed

vrundakansara commented on src/blocks/How-To/edit.js:
```diff
@@ -0,0 +1,1001 @@
+		this.props.setAttributes( { block_id: this.props.clientId } )
+
+		// Assigning block_id in the attribute.
+		this.props.setAttributes( { classMigrate: true } )
```
Why this?

AbhijitNage123 replied:
Not needed removed

vrundakansara commented on src/blocks/How-To/edit.js:
```diff
@@ -0,0 +1,1001 @@
+
+		this.props.setAttributes({ schema: JSON.stringify(this.props.schemaJsonData) });
+
+		let level_val = parseInt( this.props.attributes.headingTag.replace( 'h' , '' ) )
```
Why this?

AbhijitNage123 replied:
Updated

AbhijitNage123 replied:
I have referred advanced heading for the HTML tag. I have updated the code now. we can directly set attribute instead  of calling function.

vrundakansara commented on src/blocks/How-To/editor.scss:
```diff
@@ -0,0 +1,13 @@
+    width:10%;
+}
+
+.wp-block-uagb-how-to span.uagb-howto__time-wrap {
```
Do you think we should combine this?
```css
.wp-block-uagb-how-to span.uagb-howto__time-wrap,
.wp-block-uagb-how-to span.uagb-howto__cost-wrap {
    display: inline-flex;
}
```

AbhijitNage123 replied:
Updated

vrundakansara commented on src/blocks/How-To/styling.js:
```diff
@@ -0,0 +1,272 @@
+
+	var base_selector = `.block-editor-page #wpwrap #uagb-how-to-${ props.clientId }`
+	
+	if ( classMigrate ) {
```
Why this?

AbhijitNage123 replied:
Removed

RajkiranBagal commented on classes/class-uagb-block-helper.php:
```diff
@@ -16,6 +16,258 @@
+
+			$attr = array_merge( $defaults, $attr );
+
+			$t_selectors = "";
```
Use single quotes.

AbhijitNage123 replied:
Removed

RajkiranBagal commented on classes/class-uagb-block-helper.php:
```diff
@@ -16,6 +16,258 @@
+
+		);
+
+			$base_selector = ( $attr['classMigrate'] ) ? '.uagb-block-' : '#uagb-how-to-';
```
Do we really need this condition for this block?

AbhijitNage123 replied:
Removed

RajkiranBagal commented on src/blocks/How-To/save.js:
```diff
@@ -0,0 +1,207 @@
+
+	let url_chk = ''
+	let title = ''
+		if( typeof attributes.mainimage !== "undefined" && attributes.mainimage !== null && attributes.mainimage !=="" ){
```
Use Yoda condition checks.

AbhijitNage123 replied:
Done

RajkiranBagal commented on src/blocks/How-To/save.js:
```diff
@@ -0,0 +1,207 @@
+		
+		let url = ''
+		if( url_chk !== "" ){
+			let size = attributes.mainimage.sizes
```
Use Yoda condition checks.

AbhijitNage123 replied:
Done

RajkiranBagal commented on src/blocks/How-To/save.js:
```diff
@@ -0,0 +1,207 @@
+			let size = attributes.mainimage.sizes
+			let imageSize = attributes.imgSize
+
+			if ( typeof size !== "undefined" && typeof size[imageSize] !== "undefined" ) {
```
Use Yoda condition checks.

AbhijitNage123 replied:
Done

RajkiranBagal commented on src/blocks/How-To/styling.js:
```diff
@@ -0,0 +1,272 @@
+		imgWidth,
+		classMigrate,
+		overallAlignment,
+		headingAlign,
```
There are multiple attributes which are not used still declared, remove them

AbhijitNage123 replied:
Removed

RajkiranBagal commented:
@AbhijitNage123 Width Not working - https://imgur.com/IAtPyp5

RajkiranBagal commented:
> @AbhijitNage123
> Change this setting format to.
> Select Image
> Size
> Width

Not yet fixed

RajkiranBagal commented:
Total Time and Total cost are on same line by default - https://imgur.com/Q9qFnHD

RajkiranBagal commented:
The secondary heading color is also applying for text which is not heading - https://imgur.com/oQiPHOO

RajkiranBagal commented:
Cost and Time Margin is also applied in between - https://imgur.com/CKQxaPn
Is it intentional? Doesn't look good.

RajkiranBagal commented:
> > This should be conditional - https://imgur.com/yGFcRaU..When Time is OFF then its Margin settings should not be displayed.
> > Same for cost
> 
> Done. Added condition

Not yet fixed

RajkiranBagal commented:
> > Change these to "Tools" and "Materials" respectively - https://imgur.com/VnWwqoy
> 
> Change to Tools and Materials

Not yet fixed.

AbhijitNage123 commented:
> Do we need all these CSS files?
> 
> ```
> assets/css/blocks/how-to-tools-child.css
> assets/css/blocks/how-to-steps-child.css
> assets/css/blocks/how-to-materials-child.css
> assets/css/blocks/how-to-child.css
> assets/css/blocks/How-To.css
> ```

Removed unnecessary files

AbhijitNage123 commented:
> `dist/blocks/image/placeholder.png`
> Shift this to assets folder.

Shifted to assets

AbhijitNage123 commented:
> The secondary heading color is also applying for text which is not heading - https://imgur.com/oQiPHOO

Right. I consider it as the whole title,text of time & cost as secondary heading

AbhijitNage123 commented:
> @AbhijitNage123 Width Not working - https://imgur.com/IAtPyp5

Fixed

AbhijitNage123 commented:
> > > This should be conditional - https://imgur.com/yGFcRaU..When Time is OFF then its Margin settings should not be displayed.
> > > Same for cost
> > 
> > 
> > Done. Added condition
> 
> Not yet fixed

Fixed

AbhijitNage123 commented:
> > > Change these to "Tools" and "Materials" respectively - https://imgur.com/VnWwqoy
> > 
> > 
> > Change to Tools and Materials
> 
> Not yet fixed.

Fixed

AbhijitNage123 commented:
> > > Change these to "Tools" and "Materials" respectively - https://imgur.com/VnWwqoy
> > 
> > 
> > Change to Tools and Materials
> 
> Not yet fixed.

Fixed

RajkiranBagal commented:
@AbhijitNage123 Add relevant Notices - https://imgur.com/1HIZ14z

AbhijitNage123 commented:
> Cost and Time Margin is also applied in between - https://imgur.com/CKQxaPn
> Is it intentional? Doesn't look good.

Updated the margin setting now. margin applied to only time & cost text

AbhijitNage123 commented:
> Change block folder name from How-To to how-to

Done

AbhijitNage123 commented:
> @AbhijitNage123 Can we add a separate function for each PanelBody Settings so that it will be easier to edit in the future. Refer FAQ

Done

vrundakansara commented on src/blocks/how-to/attributes.js:
```diff
@@ -0,0 +1,382 @@
+
+	materials.push(
+		{
+			"add_required_materials": "- A WordPress Website.",
```
Translatable?

AbhijitNage123 replied:
Updated

vrundakansara commented on src/blocks/how-to/attributes.js:
```diff
@@ -0,0 +1,382 @@
+
+	tools.push(
+		{
+			"add_required_tools": "- A Computer.",
```
Translatable?

AbhijitNage123 replied:
Done

vrundakansara commented on src/blocks/how-to/editor.scss:
```diff
@@ -0,0 +1,41 @@
+.wp-block-uagb-how-to .block-editor-button-block-appender {
```
Make optimal use of scss.

AbhijitNage123 replied:
updated.

vrundakansara commented:
Remove Tool & Materials typography.
- Apply the Description typography to tools/material list.
- Apply the Secondary Heading typography to tools/material heading.

vrundakansara commented:
Have we made any use of this placeholder?
`assets/image/placeholder.png`

vrundakansara commented on classes/class-uagb-helper.php:
```diff
@@ -377,6 +377,10 @@ public function get_block_css( $block ) {
 
             switch ( $name ) {
+            	case 'uagb/how-to':
+            	    $css += UAGB_Block_Helper::get_how_to_css( $blockattr, $block_id );
```
You missed the function to enqueue the font family function. Refer Advanced Heading.

AbhijitNage123 replied:
Added.

AbhijitNage123 commented:
> Remove Tool & Materials typography.
> 
> * Apply the Description typography to tools/material list.
> * Apply the Secondary Heading typography to tools/material heading.

Done

AbhijitNage123 commented:
> Have we made any use of this placeholder?
> `assets/image/placeholder.png`

No. removed

RajkiranBagal commented on classes/class-uagb-block-js.php:
```diff
@@ -190,6 +190,34 @@ public static function blocks_advanced_heading_gfont( $attr ) {
+		/**
+		 * Adds Google fonts for How To block.
+		 *
+		 * @since 1.9.1
```
add since x.x.x

AbhijitNage123 replied:
Done

RajkiranBagal commented on classes/class-uagb-block-helper.php:
```diff
@@ -548,7 +666,7 @@ public static function get_column_css( $attr, $id ) {
 		 * @return array The Widget List.
 		 */
-		public static function get_adv_heading_css( $attr, $id ) {
+		public static function get_adv_heading_css( $attr, $id ) { // @codingStandardsIgnoreStart
```
@AbhijitNage123 These comments were removed by Vrunda Mam it seems, and now this PR is again adding those to all the blocks CSS.
Please remove these comments.

AbhijitNage123 replied:
Removed.

RajkiranBagal commented on classes/class-uagb-config.php:
```diff
@@ -84,6 +84,79 @@ public static function get_block_attributes() {
+					'uagb/how-to'             => array(
+						'slug'        => '',
+						'title'       => __( 'How-To', 'ultimate-addons-for-gutenberg' ),
+						'description' => __( 'How To Configure How-To Schema In Ultimate Addons For Gutenberg?', 'ultimate-addons-for-gutenberg' ),
```
Add appropriate Description

AbhijitNage123 replied:
I have discussed it with ma'am. will update later after discussion with Monika ma'am

RajkiranBagal commented on src/blocks/how-to/edit.js:
```diff
@@ -0,0 +1,989 @@
+const { compose } = wp.compose
+
+const {
+	registerBlockType,
```
Not used yet declared.

AbhijitNage123 replied:
Removed.done

RajkiranBagal commented on src/blocks/how-to/edit.js:
```diff
@@ -0,0 +1,989 @@
+	ToggleControl
+} = wp.components
+
+const { select, withSelect, dispatch } = wp.data;
```
select and dispatch Not used yet declared.

AbhijitNage123 replied:
Removed.done

RajkiranBagal commented on src/blocks/how-to/edit.js:
```diff
@@ -0,0 +1,989 @@
+			insertBlocksAfter,
+			mergeBlocks,
+			onReplace,
+			anchor,
```
Multiple attributes not used yet declared - https://imgur.com/ULpR9MZ

AbhijitNage123 replied:
Removed.done

RajkiranBagal commented on src/blocks/how-to/save.js:
```diff
@@ -0,0 +1,217 @@
+
+// Import block dependencies and components.
+import classnames from "classnames"
+import renderSVG from "../../../dist/blocks/uagb-controls/renderIcon"
```
Not used yet declared

AbhijitNage123 replied:
Removed.done

RajkiranBagal commented on src/blocks/how-to/save.js:
```diff
@@ -0,0 +1,217 @@
+		estCost,
+		cost,
+		currencyType,
+		hideLabel,
```
Multiple attributes not used yet declared - https://imgur.com/u3zZ3PQ

AbhijitNage123 replied:
Removed.done

RajkiranBagal commented on src/blocks/how-to/styling.js:
```diff
@@ -0,0 +1,312 @@
+	const {
+		imgWidth,
+		overallAlignment,
+		headingAlign,
```
Multiple attributes not used yet declared - https://imgur.com/qg60vs3

AbhijitNage123 replied:
Removed.done

RajkiranBagal commented:
@AbhijitNage123 Add recommended notices for these also - https://imgur.com/FRoYxeF

RajkiranBagal commented:
@AbhijitNage123  Cannot find an option to manage this GAP - https://imgur.com/S9gylLx
A Row Gap setting is required it seems

AbhijitNage123 commented:
> @AbhijitNage123 Add recommended notices for these also - https://imgur.com/FRoYxeF

Added notice.

vrundakansara commented on src/blocks/how-to/style.scss:
```diff
@@ -0,0 +1,33 @@
+.wp-block-uagb-how-to {
+	
+	.block-editor-button-block-appender {
```
Move this to editor.scss. We don't need editor code in style.scss

AbhijitNage123 replied:
Done

AbhijitNage123 commented:
> @AbhijitNage123 Cannot find an option to manage this GAP - https://imgur.com/S9gylLx
> A Row Gap setting is required it seems

Added
